### DISCUSSION TITLE
Duvida sobre codigo

### DISCUSSION THREAD

**Discussion by Arllei01** [balta-io/forum#33] [Tira Dúvidas] [answered]
Boa Tarde!
No curso de Fundamentos em C# > Linguagem em programação com C# > Métodos e funções: Prática, nesse vídeo em questão vc fez criou o seguinte método:

``` csharp
static string RetornaNome( string nome, string sobrenome, int idade)
{
     return nome + " " + sobrenome + " tem " + idade.ToString();
}
``` 
e invocou ela no método Main e fez um WriteLine.

A minha duvida é: Você colocou um **int** no método **RentornaNome** e no  **return** vc fez a conversão de um **int** para uma **string** usando o **ToString()**, porém no meu código eu sem querer acabei esquecendo de utilizar o **ToString()** e mesmo assim funcionou normalmente. O fato do meu condigo funcionar sem a conversão é devido a alguma atualização do C# que converte automático? ou funcionou por algum outro motivo, ou não deveria funcionar? 

Obs: Não sei a data de criação do vídeo, eu estou utilizando a versão mais recendo do VSCode.

Seus vídeos são ótimos, bem claros na explicação e bem intuitivo, Parabéns!!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Arllei01, como vai?

É uma ótima pergunta 😄

Isto ocorre por que todo tipo no .NET deriva de um tipo base que possui o método `ToString()`.
Desta forma, sempre que temos uma propriedade ou objeto em uma string, este método já é invocado.

Ou seja, neste caso, tanto faz utilizar o `.ToString()` pois você já está em uma string (Concatenando os valores), então o C# é inteligente o bastante para entender isto e invocar o método automaticamente.

Porém no curso eu preciso explicitar para você saber o que está acontecendo "por baixo dos panos". Mais adiante, nos próximos cursos, eu mostro bem mais coisas sobre isto.

**Reply by Arllei01** [reply]
Entendi.
Muito obrigado!
